mono/metadata/threads.c

```diff
@@ -1639,11 +1640,13 @@ ves_icall_System_Threading_Thread_Join_internal(MonoThread *this_obj, int ms)
 	mono_thread_set_state (cur_thread, ThreadState_WaitSleepJoin);
 
 	MONO_ENTER_GC_SAFE;
-	ret=mono_thread_info_wait_one_handle (handle, ms, TRUE);
+	ret=mono_join_uninterrupted (handle, ms, &error);
```

Comment:
I'm pretty sure this broke cooperative GC mode - the MONO_{ENTER,EXIT}_GC_SAFE pair around this will switch the thread to blocking state, which will cause an illegal coop state transition when mono_join_uninterrupted also call MONO_ENTER_GC_SAFE in threads.c:1728